ee/vellum_cli/push.py

```diff
@@ -130,12 +130,22 @@ def push_command(
     )
     sys.path.insert(0, os.getcwd())
 
+    # Load metadata.json to get trigger_path_to_id_mapping
+    trigger_path_to_id_mapping = {}
+    module_dir = workflow_config.module.replace(".", os.path.sep)
+    metadata_path = os.path.join(module_dir, "metadata.json")
+    if os.path.exists(metadata_path):
+        with open(metadata_path) as f:
+            metadata_json = json.load(f)
+            trigger_path_to_id_mapping = metadata_json.get("trigger_path_to_id_mapping", {})
+
     # Remove this once we could serialize using the artifact in Vembda
     # https://app.shortcut.com/vellum/story/5585
     serialization_result = BaseWorkflowDisplay.serialize_module(
         workflow_config.module,
         client=client,
         dry_run=dry_run or False,
+        trigger_path_to_id_mapping=trigger_path_to_id_mapping,
```

Comment:
[blocking] Oh this might scale poorly and involve a kwarg for every kind of id mapping. If we already know how to get the mapping based on `workflow_config.module`, then this kwarg is redundant and could be calculated internally from `workflow_config.module` instead